src/vorta/views/archive_tab.py

```diff
@@ -488,15 +492,28 @@ def list_result(self, result):
             self.populate_from_profile()
 
     def refresh_archive_info(self):
-        archive_name = self.selected_archive_name()
-        if archive_name is not None:
-            params = BorgInfoArchiveJob.prepare(self.profile(), archive_name)
-            if params['ok']:
-                job = BorgInfoArchiveJob(params['cmd'], params, self.profile().repo.id)
-                job.updated.connect(self._set_status)
-                job.result.connect(self.info_result)
-                self._toggle_all_buttons(False)
-                self.app.jobs_manager.add_job(job)
+        selected_archives = self.archiveTable.selectionModel().selectedRows()
+        profile = self.profile()
+
+        name_list = []
+        for index in selected_archives:
+            name_list.append(self.archiveTable.item(index.row(), 4).text())
+
+        archive_list = (
+            profile.repo.archives.select().where(ArchiveModel.name << name_list).order_by(ArchiveModel.time.desc())
+        )
+
+        archive_names = [archive.name for archive in archive_list]
+
+        for archive_name in archive_names:
+            if archive_name is not None:
+                params = BorgInfoArchiveJob.prepare(self.profile(), archive_name)
+                if params['ok']:
+                    job = BorgInfoArchiveJob(params['cmd'], params, self.profile().repo.id)
+                    job.updated.connect(self._set_status)
+                    job.result.connect(self.info_result)
+                    self._toggle_all_buttons(False)
+                    self.app.jobs_manager.add_job(job)
 
     def info_result(self, result):
         self._toggle_all_buttons(True)
```

Comment:
This will enable the buttons again although the other info jobs are still running.

Reply:
So should I connect just the last job's result to this function?

Reply:
Well, theoretically you don't know which job finishes first. Did you try to run the current code? I would think that only one borg jobs succeeds because of the repo lock.

Reply:
Yeah I tried running it (I attached a GIF above).

Reply:
Nevermind, I didn't attach the GIF. Here,
![Peek 2023-06-15 02-22](https://github.com/borgbase/vorta/assets/41837037/d810ea57-1dd1-483b-a4f9-9c24d63c2d6b)

They do blink.